pkg/flow/internal/controller/loader.go

```diff
@@ -435,7 +435,7 @@ func (l *Loader) populateComponentNodes(g *dag.Graph, componentBlocks []*ast.Blo
 		blockMap = make(map[string]*ast.BlockStmt, len(componentBlocks))
 	)
 	for _, block := range componentBlocks {
-		var c *ComponentNode
+		var c *BuiltinComponentNode
```

Comment:
It seems likely that we'll want this to be the interface instead of the concrete type:

```suggestion
		var c ComponentNode
```